src/HucaresServer.Storage.UnitTests/DetectedPlateHelperTests.cs

```diff
@@ -120,37 +120,63 @@ public void InsertNewDetectedPlate_WhenUriNotCorrect_ShouldThrowError()
         public void GetAllDetectedPlates_ShouldReturn()
         {
             //Arrange
-            var fakeList = new List<DetectedLicensePlate>(){
+            
+            var expectedDetectedPlate = new DetectedLicensePlate()
+            {
+                Id = 1, PlateNumber = "ABC002", DetectedDateTime = new DateTime(2018, 09, 30), 
+                CamId = 2, ImgUrl = "http://localhost:6969/images", Confidence = 0.80
+            };
+            
+            var fakeDetectedPlateList = new List<DetectedLicensePlate>(){
+                expectedDetectedPlate,
                 new DetectedLicensePlate()
                 {
                     Id = 0, PlateNumber = "ABC001", DetectedDateTime = new DateTime(2018, 09, 29), 
                     CamId = 1, ImgUrl = "http://localhost:6969/images", Confidence = 0.75
                 },
                 new DetectedLicensePlate()
                 {
-                    Id = 1, PlateNumber = "ABC002", DetectedDateTime = new DateTime(2018, 09, 30), 
+                    Id = 2, PlateNumber = "ABC003", DetectedDateTime = new DateTime(2018, 09, 30), 
                     CamId = 2, ImgUrl = "http://localhost:6969/images", Confidence = 0.80
                 }
             };
             
-            var fakeDbSet = StorageTestsUtil.SetupFakeDbSet(fakeList.AsQueryable());
+            var fakeMissingPlateList = new List<MissingLicensePlate>()
+            {
+                new MissingLicensePlate()
+                {
+                    Id = 0, 
+                    PlateNumber = expectedDetectedPlate.PlateNumber,
+                    SearchStartDateTime = new DateTime(2018, 09, 29),
+                    SearchEndDateTime = null, 
+                    LicensePlateFound = false
+                }
+            };
 
+            var fakeDbSetMissingPlates = StorageTestsUtil.SetupFakeDbSet((fakeMissingPlateList.AsQueryable()));
+            var fakeDbSetDetectedPlates = StorageTestsUtil.SetupFakeDbSet(fakeDetectedPlateList.AsQueryable());
             var fakeHucaresContext = A.Fake<HucaresContext>();
+            
             A.CallTo(() => fakeHucaresContext.DetectedLicensePlates)
-                .Returns(fakeDbSet);
+                .Returns(fakeDbSetDetectedPlates);
 
+            A.CallTo(() => fakeHucaresContext.MissingLicensePlates)
+                .Returns(fakeDbSetMissingPlates);
+            
             var fakeDbContextFactory = A.Fake<IDbContextFactory>();
             A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
                 .Returns(fakeHucaresContext);
             
-            var detectedPlateHelper = new DetectedPlateHelper(fakeDbContextFactory);
+            var missingPlateHelper = new MissingPlateHelper(fakeDbContextFactory);
```

Comment:
This is wrong. When you do this, you are depending on the implementation code written in missingPlateHelper, and as such this unit test is no longer isolated to just your class. Instead of faking out the whole implamentation of the class, do
A.Fake<IMissingPlateHelper>(); and simply state that a call to GetAllRecords on this fake dependency returns the list you want it to return.